hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieDataBlock.java

```diff
@@ -84,19 +86,32 @@ public HoodieDataBlock(List<HoodieRecord> records,
                          Map<FooterMetadataType, String> footer,
                          String keyFieldName) {
     super(header, footer, Option.empty(), Option.empty(), null, false);
-    if (shouldWriteRecordPositions) {
-      records.sort((o1, o2) -> {
-        long v1 = o1.getCurrentPosition();
-        long v2 = o2.getCurrentPosition();
-        return Long.compare(v1, v2);
-      });
-      if (isPositionValid(records.get(0).getCurrentPosition())) {
-        addRecordPositionsToHeader(
-            records.stream().map(HoodieRecord::getCurrentPosition).collect(Collectors.toSet()),
-            records.size());
+    if (shouldWriteRecordPositions && !records.isEmpty()) {
+      String baseFileInstantTime = records.get(0).getCurrentBaseFileInstantTime();
+      for (HoodieRecord record : records) {
+        if (!baseFileInstantTime.equals(record.getCurrentBaseFileInstantTime())) {
+          baseFileInstantTime = EMPTY_STRING;
+          break;
+        }
+      }
+      if (StringUtils.isNullOrEmpty(baseFileInstantTime)) {
+        LOG.warn("Base file instant time of delete positions cannot be determined, possibly "
```

Comment:
fix logging. this is not delete blocks. why talk about delete positions

Reply:
Fixed.